pages/DynamicPOC/index.tsx

```diff
@@ -75,47 +81,41 @@ const DynamicPOC: React.FC<{ questions: Question[] }> = ({ questions }) => {
   const [questionHistory, setQuestionHistory] = useState([startingQuestion])
   const [currentIndex, setCurrentIndex] = useState(0)
 
-  function getNextQuestion(answer: Answer): Question {
-    const id: number = answer.route
-    return questions[id]
+  /**
+   * Returns the next question based on whether or not current question is a radio, continue, or text,
+   * taking advantage that continue and text qs only have one route after (or should..)
+   */
+  function getNextQuestion(q: Question, a: FormAnswer): Question {
+    return questions[
+      q.answers[a.type === QuestionType.RADIO ? a.answerId : 0].route
+    ]
   }
 
-  function _updateCurrentAnswer(
-    questionId: string,
-    answerId: string,
-    userAnswer?: string
-  ) {
-    const answer = {
-      questionId,
-      answerId,
-      userAnswer,
+  /**
+   * Handles changing the old question to the given question
+   */
+  function _handleQuestionChange(nextQuestion: Question) {
+    formValues.formAnswers[currentQuestion.id] = currentAnswer
```

Comment:
General question - is there a reason we don't manipulate formValues with useState/setState? I know it has something to do with the form context but would love some more clarification.

Reply:
Oh yeah so formValues isn't actually state since a change to formValues doesn't really affect the way the current screen is rendered. The useContext (from what I understand) is how we give formik the current form information and whatnot

Reply:
Would we be able to do a `useEffect` with formValues, or no? Trying to look ahead to connecting form saving frontend to the endpoint.

Reply:
Definitely possible but I do think that's beyond the scope of this ticket